File: app/src/main/java/com/juniperphoton/myersplash/model/UnsplashUser.java
package com.juniperphoton.myersplash.model;

import com.google.gson.annotations.SerializedName;

import java.io.Serializable;

public class UnsplashUser implements Serializable {
    @SerializedName("id")
    private String mId;

    @SerializedName("username")
    private String mUserName;

    @SerializedName("name")
    private String mName;

    @SerializedName("links")
    private ProfileUrl mLinks;

    public UnsplashUser() {

    }

    public String getName() {
        return mName;
    }

    public String getHomeUrl() {
        return mLinks.mHtml;
    }

    public static class ProfileUrl implements Serializable {
        @SerializedName("self")
        private String mSelf;

        @SerializedName("html")
        private String mHtml;

        @SerializedName("photos")
        private String mPhotos;

        @SerializedName("likes")
        private String mLikes;

        @SerializedName("portfolio")
        private String mPortfolio;

        public ProfileUrl() {

        }
    }
}

File: app/src/main/kotlin/com/juniperphoton/myersplash/model/UnsplashUser.kt
package com.juniperphoton.myersplash.model

import com.google.gson.annotations.SerializedName

import java.io.Serializable

@Suppress("unused")
class UnsplashUser : Serializable {
    @SerializedName("id")
    var id: String? = null

    @SerializedName("username")
    var userName: String? = null

    @SerializedName("name")
    var name: String? = null

    @SerializedName("links")
    var links: ProfileUrl? = null

    val homeUrl: String?
        get() = links?.html
}

@Suppress("unused")
class ProfileUrl : Serializable {
    @SerializedName("self")
    val self: String? = null

    @SerializedName("html")
    var html: String? = null

    @SerializedName("photos")
    val photos: String? = null

    @SerializedName("likes")
    val likes: String? = null

    @SerializedName("portfolio")
    val portfolio: String? = null
}